cloud.foundry.cli/src/main/java/cloud/foundry/cli/operations/ApplicationsOperations.java

```diff
@@ -123,6 +117,32 @@ public ApplicationsOperations(DefaultCloudFoundryOperations cloudFoundryOperatio
         }
     }
 
+    /**
+     * Prepares a request for renaming an application instance.
+     * The resulting mono is preconfigured such that it will perform logging.
+     *
+     * @param newName     new name of the application instance
+     * @param currentName current name of the application instance
+     * @return mono which can be subscribed on to trigger the renaming request to the cf instance
+     * @throws NullPointerException when one of the arguments was null
+     */
+    public Mono<Void> rename(String newName, String currentName) {
+        checkNotNull(newName);
+        checkNotNull(currentName);
+
+        RenameApplicationRequest renameApplicationRequest = RenameApplicationRequest.builder()
+                .name(currentName)
+                .newName(newName)
+                .build();
+
+        return this.cloudFoundryOperations.applications().rename(renameApplicationRequest)
+                .doOnSubscribe(aVoid -> {
+                    log.debug("Rename application:", currentName);
+                    log.debug("With new name:", newName); })
+                .doOnSuccess(aVoid -> log.info("Application renamed from", currentName, "to", newName))
+                .onErrorStop();
```

Comment:
You should describe int the documentation, that the process will stop, when an error occurs.

Reply:
Is the .onErrorStop() call even necessary? The rename request is just a single mono with no involved chaining via a flux

Reply:
No it is not. I have removed it.